src/mappings/events/vaults.ts

```diff
@@ -53,8 +53,84 @@ export const vaultsEventKit = (block: any, data: any, module: string, path: stri
     return promises;
   }
 
+  async function updateDailyVaultState(
+    oldState: string | undefined,
+    newState: string,
+    blockTime: Date,
+    blockHeight: number,
+  ): Promise<VaultStatesDaily> {
+    const dateKey = dateToDayKey(blockTime).toString();
+    let vaultState: VaultStatesDaily | undefined = await VaultStatesDaily.get(dateKey);
+
+    if (!vaultState) {
+      const yesterdayDate = new Date(blockTime);
+      yesterdayDate.setDate(yesterdayDate.getDate() - 1);
+      const yesterdayDateKey = dateToDayKey(yesterdayDate).toString();
+      const yesterdayVaultState: VaultStatesDaily | undefined = await VaultStatesDaily.get(yesterdayDateKey);
+
+      if (yesterdayVaultState) {
+        vaultState = new VaultStatesDaily(
+          dateKey,
+          BigInt(blockHeight),
+          blockTime,
+          yesterdayVaultState.active,
+          yesterdayVaultState.closed,
+          yesterdayVaultState.liquidating,
+          yesterdayVaultState.liquidated,
+          yesterdayVaultState.liquidatedClosed,
+        );
+      } else {
+        vaultState = new VaultStatesDaily(
+          dateKey,
+          BigInt(blockHeight),
+          blockTime,
+          BigInt(0),
+          BigInt(0),
+          BigInt(0),
+          BigInt(0),
+          BigInt(0),
+        );
+      }
+    }
+
+    const propertyMap = {
+      [VAULT_STATES.ACTIVE]: 'active',
+      [VAULT_STATES.LIQUIDATED]: 'liquidated',
+      [VAULT_STATES.LIQUIDATING]: 'liquidating',
+      [VAULT_STATES.CLOSED]: 'closed',
+    };
+
+    const closedPropertyMap = {
+      [VAULT_STATES.ACTIVE]: 'closed',
+      [VAULT_STATES.LIQUIDATED]: 'liquidatedClosed',
+    };
+
+    if (oldState) {
+      const oldProperty = propertyMap[oldState];
+      if ((vaultState as any)[oldProperty] === BigInt(0)) {
+        throw Error(oldState + ' vaults are 0. cannot subtract more');
```

Comment:
Why `as` assertion? Why not convert `oldProperty` to big int?

Reply:
the issue here is that typescript warns me about indexing a class object with a string. which is physically possible but the type wont allow it.
therefore i have to convert the objects type to any before indexing it

Reply:
Goal is to compare big int with big int? this doesn't work? `if (BigInt(oldProperty) === BigInt(0)) `?

Reply:
oldProperty is the name of the propert e.g "active", "liquidated" etc

we index it from "vaultState" to get the actual BigInt value